At modify

lainsworth8801 opened the pull request:
Branched off Master. Modified geojson_to_buildingsync to generate matching format for audit template; utility script buildingsync_at_modify for existing BuildingSync files from seed for audit template. Error fixes in progress.

Commit:
BuildingSync file modification for audit template; geojson to generate new format for audit template

Commit:
Removed dead code

Commit:
Moved/Added street address

lainsworth8801 committed:
More modification for latest buildingsync from seed

lainsworth8801 committed:
XPath to return building id for rp report

lainsworth8801 committed:
Clean up

lainsworth8801 committed:
More code cleanup

lainsworth8801 committed:
Changed warning message for missing elements

nllong commented on bin/buildingsync_at_modify.rb:
```diff
@@ -0,0 +1,108 @@
+  tag = Hash.new
+
+  # find line number of <auc:Address> and </auc:Ownership>
+  File.foreach(file).with_index(1) do |line, index|
```
this assumes that the XML has been formated "nicely"

lainsworth8801 replied:
yes. I'm trying to convert this part to using Xpath to insert elements. But BSync files are formatted nicely from seed anyway which I think is what imma did unless I manually download 'result.xml'

nllong replied:
Hmm, i just picked a random file that this has to run against and it doesn't look "nicely" formatted.  File was "buildingsync_files/9473.xml". I will test locally.

nllong commented on bin/buildingsync_at_modify.rb:
```diff
@@ -0,0 +1,108 @@
+
+      # Remove all start and end timestamp. Standard xml for JSON
+      doc = REXML::Document.new File.new(file)
+      doc.elements.delete_all("//auc:StartTimeStamp")
```
is there a reason we are removing these and not just renaming them to the new format (which should be StartTimestamp)?

lainsworth8801 replied:
No idea. this was my question this morning. AT complained every single one of them and would let import proceed once I removed them. The thing is in both of their sample files starttimestamp and endtimestamp are allowed. I sent out an email on Friday to David Millard but haven't heard back anything.

nllong replied:
If the files successfully roundtrip then I think we can ignore this... otherwise the change (i think) is that `StartTimeStamp -> StartTimestamp` and `EndTimeStamp -> EndTimestamp`.

lainsworth8801 replied:
You are right. Did not notice that lower case 's'.

nllong commented on bin/buildingsync_at_modify.rb:
```diff
@@ -0,0 +1,108 @@
+  end
+end
+
+modify_existing_bs_file()
```
put a hard return on the file.

nllong commented on geojson_to_buildingsync.rb:
```diff
@@ -1026,15 +1027,30 @@ def convert_feature(feature)
 
-  source += '      </auc:Scenarios>
-        </auc:Report>
+  source += "      </auc:Scenarios>
```
This is okay for now, but I think we want to convert this over to REXML::Element.new type of code.

lainsworth8801 replied:
Also working on this. At the time I had not discovered rexml.

nllong commented on bin/buildingsync_at_modify.rb:
```diff
@@ -0,0 +1,108 @@
+
+  # find line number of <auc:Address> and </auc:Ownership>
+  File.foreach(file).with_index(1) do |line, index|
+    (0...list.size).each { |i|
```
ruby likes the `do` command instead of the curly braces `{`, `}`

nllong commented on bin/buildingsync_at_modify.rb:
```diff
@@ -0,0 +1,108 @@
+      end
+
+      remove = ownership_index - addr_index + 1
+      (addr_index..ownership_index).each {|line| content += File.readlines(file)[line - 1]
```
if this is a single line then curly braces are "okay".

nllong reviewed (changes requested):
Most of these are notes. The only one I am concerned about is the `doc.elements.delete_all("//auc:StartTimeStamp")`

lainsworth8801 committed:
Ugly code. Need cleanup;
xpath add elements; CData to add report; changed start/endtimestamp to lower case 'stamp'

lainsworth8801 committed:
Corrected file path

lainsworth8801 committed:
Filter correct building id

lainsworth8801 committed:
write correct format

lainsworth8801 committed:
BSync file path

lainsworth8801 committed:
Multi-thread, code cleanup

lainsworth8801 committed:
Removed dead code; puts file basename, added move old schema files to separate dir

lainsworth8801 committed:
Added rake task to run buildingsync_to_at.rb on command line.

lainsworth8801 committed:
Added double quote for building ID. More cleanup

lainsworth8801 committed:
Add empty line for hard return
